### ISSUE TITLE
[FloatingActionButton] App crashes because of shapeAppearance on FAB

### ISSUE THREAD

**Issue by ashraf-patel** [material-components/material-components-android#661]
**Description:** Here is the Demo app https://github.com/ashraf-patel/MaterialComponentsDemo, as described in this [link to the gif](https://github.com/ashraf-patel/MaterialComponentsDemo/blob/master/gifs/30dpCorners.gif) you can reproduce the crash. 

**Expected behavior:** It should not crash.

**Source code:** 

Here is the problem, making  cornerSizeTopRight and cornerSizeBottomLeft to 30dp is causing the crash. 
[link to the Gif](https://github.com/ashraf-patel/MaterialComponentsDemo/blob/master/gifs/30dpCorners.gif)
```
<style name="ShapeAppearance.Sunflower.FAB" parent="ShapeAppearance.MaterialComponents">
        <item name="cornerFamily">rounded</item>
        <item name="cornerSizeTopLeft">0dp</item>
        <item name="cornerSizeTopRight">30dp</item>
        <item name="cornerSizeBottomRight">0dp</item>
        <item name="cornerSizeBottomLeft">30dp</item>
</style>
```
Below code works [link to the gif](https://github.com/ashraf-patel/MaterialComponentsDemo/blob/master/gifs/0dpCorners.gif)
```
<style name="ShapeAppearance.Sunflower.FAB" parent="ShapeAppearance.MaterialComponents">
        <item name="cornerFamily">rounded</item>
        <item name="cornerSizeTopLeft">0dp</item>
        <item name="cornerSizeTopRight">0dp</item>
        <item name="cornerSizeBottomRight">0dp</item>
        <item name="cornerSizeBottomLeft">0dp</item>
</style>
``` 
https://github.com/ashraf-patel/MaterialComponentsDemo is the Demo app.

**Android API version:** OS: 8.0

**Material Library version:** 1.1.0-beta01

**Device:** Device: Pixel 2 XL Emulator

**StackOverflow:** https://stackoverflow.com/questions/58225921/shapeappearance-in-material-component-floatingactionbutton-causes-a-crash

**Issue in original repository:** https://github.com/android/sunflower/issues/519

**Comment by ymarian** [comment]
30dp is bigger than the radius of the fab. A normal fab is 56dp. 
Either set: 

`<dimen name="fab_corner_radius">50%</dimen>`

If you do want a 60dp fab you can also set:

`app:fabCustomSize="60dp"`

Depending on what you are trying to achieve.

**Comment by ashraf-patel** [comment]
`<dimen name="fab_corner_radius">50%</dimen>` still crashes.
But this `<dimen name="fab_corner_radius">28dp</dimen>` works with default FAB size.
Also `<dimen name="fab_corner_radius">30dp</dimen>` works with `app:fabCustomSize="60dp"`

**Comment by ashraf-patel** [comment]
Any idea why `<dimen name="fab_corner_radius">50%</dimen>` is not working here?

**Comment by gabrielemariotti** [comment]
@ymarian 
Just a curiosity.
Using the `app:shapeAppearanceOverlay` with 2 corners of `30dp` as described in the issue in a standard fab (`56dp`):

 ```
<style name="CustomShapeOVerlay" parent="">
        <item name="cornerFamily">rounded</item>
        <item name="cornerSizeTopLeft">0dp</item>
        <item name="cornerSizeTopRight">30dp</item>
        <item name="cornerSizeBottomRight">0dp</item>
        <item name="cornerSizeBottomLeft">30dp</item>
  </style>
```
It works with 1.1.0-beta01 in devices with API28 and API29.

<img width="58" alt="AA9aQ" src="https://user-images.githubusercontent.com/2583078/66432394-3c1d9180-ea1e-11e9-90e6-3184d035bcaa.png">

**Comment by ymarian** [comment]
My bad the corner can be more than 50% since the other one on the same side is 0.

@gabrielemariotti It repros in 26. 

I'll have to debug this tomorrow setting app:shapeAppearanceOverlay="@null" seems to help, which doesn't make sense to me since the ShapeAppearance in the example is setting the more specific cornerSizes. The default fab overlay only sets `app:cornerSize` to 50% the shape appearance should ignore the overlay in this case.

**Comment by ashraf-patel** [comment]
@ymarian any updates?

**Comment by cketcham** [comment]
Looks like a native crash. I was only able to repro it on API 26 emulator. I haven't tried on a real device. Are we sure this crashes in that case too?

**Comment by ashraf-patel** [comment]
Do you mean on real device?

**Comment by cketcham** [comment]
Yeah I was wondering if you've tested on a real device. But after a bit of digging around, it looks like this is an issue on API 26, and I saw some similar reports coming from real devices in other cases.

We found a workaround to not set the scale close to 0 while the fab is animating on API 26. That seems to fix it. We'll sync out the fix to github soon.

**Comment by cketcham** [comment]
Resolved by 3c3ac61
